app/views/ontologies/new.html.haml

```diff
@@ -23,18 +23,18 @@
                   .upload-ontology-desc
                     = metadata_help_link
                   .upload-ontology-input-field-container
-                    = attribute_input('URI', label: t("ontologies.uri"))
+                    = attribute_input('URI', label: t("ontologies.uri"), value: params.dig(:submission, :URI))
```

Comment:
you don't need to add `value:  params.dig(:submission, :URI)`, because `attribute_input` will automatically fetch as value `@submission.send(your_key)` in this example here will call automatically  `@submission.send(:URI)`

Reply:
same for all the usage of the helper `attribute_input` in this page, the solution is not to add the argument `value`, but is to fix `attribute_input` so that it fetches automatically the correct value depending of the attribute key

Reply:
the values are fetched here 
https://github.com/ontoportal-lirmm/bioportal_web_ui/blob/7a22a9f4c0dc3b189d549ba064928912b85cee0b/app/helpers/submission_inputs_helper.rb#L27-L32

and used when calling `attr.values `in the code, when needed, search for its occurrence to see its usage

Reply:
When the user submit a form that has errors, we don't save the submission!
in all cases we need to use the params to get bach the values that the user entered in the form

Reply:
is `@submssion` empty ?, and you can use params, as it will not work in the add new submission version (not the ontology creation version) 

Any way `@submission` should already be filled with the params, even if not saved.

Reply:
You don't have `params` in the "add submission" version, go to an existent ontology and click on "add a new submission", `params` will be empty, so the initial state will be empty.

Reply:
`attribute_input` is also used on the submission edit page, check also that if you submit the wrong metadata, it shows the error page, and with the values filled.